staging/src/k8s.io/cloud-provider/controllers/service/controller_test.go

```diff
@@ -145,160 +178,63 @@ func TestSyncLoadBalancerIfNeeded(t *testing.T) {
 			expectOp:             ensureLoadBalancer,
 			expectCreateAttempt:  true,
 			expectPatchStatus:    true,
 			expectPatchFinalizer: true,
 		},
 		{
-			desc: "tcp service that wants LB",
-			service: &v1.Service{
-				ObjectMeta: metav1.ObjectMeta{
-					Name:      "basic-service1",
-					Namespace: "default",
-				},
-				Spec: v1.ServiceSpec{
-					Ports: []v1.ServicePort{{
-						Port:     80,
-						Protocol: v1.ProtocolTCP,
-					}},
-					Type: v1.ServiceTypeLoadBalancer,
-				},
-			},
+			desc:                 "tcp service that wants LB",
+			service:              newService("basic-service1", "", v1.ServiceTypeLoadBalancer, tweakAddPorts(v1.ProtocolTCP, "")),
 			expectOp:             ensureLoadBalancer,
 			expectCreateAttempt:  true,
 			expectPatchStatus:    true,
 			expectPatchFinalizer: true,
 		},
 		{
-			desc: "sctp service that wants LB",
-			service: &v1.Service{
-				ObjectMeta: metav1.ObjectMeta{
-					Name:      "sctp-service",
-					Namespace: "default",
-				},
-				Spec: v1.ServiceSpec{
-					Ports: []v1.ServicePort{{
-						Port:     80,
-						Protocol: v1.ProtocolSCTP,
-					}},
-					Type: v1.ServiceTypeLoadBalancer,
-				},
-			},
+			desc:                 "sctp service that wants LB",
+			service:              newService("sctp-service", "", v1.ServiceTypeLoadBalancer, tweakAddPorts(v1.ProtocolSCTP, "")),
 			expectOp:             ensureLoadBalancer,
 			expectCreateAttempt:  true,
 			expectPatchStatus:    true,
 			expectPatchFinalizer: true,
 		},
 		{
-			desc: "service specifies loadBalancerClass",
-			service: &v1.Service{
-				ObjectMeta: metav1.ObjectMeta{
-					Name:      "with-external-balancer",
-					Namespace: "default",
-				},
-				Spec: v1.ServiceSpec{
-					Type:              v1.ServiceTypeLoadBalancer,
-					LoadBalancerClass: utilpointer.StringPtr("custom-loadbalancer"),
-				},
-			},
+			desc:                 "service specifies loadBalancerClass",
+			service:              newService("with-external-balancer", "", v1.ServiceTypeLoadBalancer, tweakAddLBC(utilpointer.StringPtr("custom-loadbalancer"))),
 			expectOp:             deleteLoadBalancer,
 			expectCreateAttempt:  false,
 			expectPatchStatus:    false,
 			expectPatchFinalizer: false,
 		},
 		{
-			desc: "service doesn't specify loadBalancerClass",
-			service: &v1.Service{
-				ObjectMeta: metav1.ObjectMeta{
-					Name:      "with-external-balancer",
-					Namespace: "default",
-				},
-				Spec: v1.ServiceSpec{
-					Ports: []v1.ServicePort{{
-						Port:     80,
-						Protocol: v1.ProtocolSCTP,
-					}},
-					Type:              v1.ServiceTypeLoadBalancer,
-					LoadBalancerClass: nil,
-				},
-			},
+			desc:                 "service doesn't specify loadBalancerClass",
+			service:              newService("with-external-balancer", "", v1.ServiceTypeLoadBalancer, tweakAddLBC(nil), tweakAddPorts(v1.ProtocolSCTP, "")),
```

Comment:
class being nil is the default, so this is identical to the case "sctp service that wants LB" above?

Reply:
IMHO, they are both the same. so I remove this case